### PR TITLE
Reusable Cache types

### PR DESCRIPTION
**Pull request by kevgo**
Eliminating some code duplication from the various caches.

### CODE REVIEW

File: src/git/cache.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+// Set allows collaborators to signal when the current branch has changed.
+func (sc *StringCache) Set(newValue string) {
+	sc.value = newValue
+	sc.initialized = true
+}
+
+// Value provides the current value.
+func (sc *StringCache) Value() string {
+	if !sc.initialized {
+		panic("using current branch before initialization")
```
Comment (charlierudolph) [inline src/git/cache.go:19]: This error message seems wrong. Same for string slice.

File: src/git/prod_repo.go [hunk 2]
```diff
@@ -14,12 +14,26 @@ type ProdRepo struct {
 func NewProdRepo() *ProdRepo {
 	silentShell := command.SilentShell{}
 	config := Config()
-	currentBranchTracker := CurrentBranchCache{}
-	remoteBranchCache := RemoteBranchCache{}
-	remotesCache := RemotesCache{}
-	silentRunner := NewRunner(silentShell, config, &currentBranchTracker, &remotesCache, &remoteBranchCache)
+	currentBranchTracker := StringCache{}
+	remoteBranchCache := StringSliceCache{}
+	remotesCache := StringSliceCache{}
```
Comment (charlierudolph) [inline src/git/prod_repo.go:19]: Nice simplification

File: src/git/cache.go (hunk 1 above)
Comment (kevgo) [reply src/git/cache.go:19]: Thanks, good catch!

**Comment by kevgo** [comment]
Thanks for the helpful reviews!

**Pull request merged**